src/openvic-simulation/modifier/StaticModifierCache.cpp

```diff
@@ -1,115 +1,166 @@
+	total_occupation {"total_occupation", {}, STATIC},
+	total_blockaded {"total_blockaded", {}, STATIC},
 	in_bankruptcy { nullptr },
+	bad_debtor { nullptr },
 	// Province modifiers
-	overseas { nullptr },
-	coastal { nullptr },
-	non_coastal { nullptr },
-	coastal_sea { nullptr },
-	sea_zone { nullptr },
-	land_province { nullptr },
-	blockaded { nullptr },
-	no_adjacent_controlled { nullptr },
-	core { nullptr },
-	has_siege { nullptr },
-	occupied { nullptr },
-	nationalism { nullptr },
-	infrastructure { nullptr } {}
+	overseas {"overseas", {}, STATIC},
+	coastal {"coastal", {}, STATIC},
+	non_coastal {"non_coastal", {}, STATIC},
+	coastal_sea {"coastal_sea", {}, STATIC},
+	sea_zone {"sea_zone", {}, STATIC},
+	land_province {"land_province", {}, STATIC},
+	blockaded {"blockaded", {}, STATIC},
+	no_adjacent_controlled {"no_adjacent_controlled", {}, STATIC},
+	core {"core", {}, STATIC},
+	has_siege {"has_siege", {}, STATIC},
+	occupied {"occupied", {}, STATIC},
+	nationalism {"nationalism", {}, STATIC},
+	infrastructure {"infrastructure", {}, STATIC} {}
 
-bool StaticModifierCache::load_static_modifiers(ModifierManager& modifier_manager) {
+bool StaticModifierCache::load_static_modifiers(ModifierManager& modifier_manager,ast::NodeCPtr const& root) {
 	bool ret = true;
+	key_map_t key_map {};
 
-	const auto set_static_modifier = [&modifier_manager, &ret](
-		Modifier const*& modifier, std::string_view name, fixed_point_t multiplier = 1
+	const auto set_static_country_modifier = [&key_map, &modifier_manager, &ret](
+		Modifier& modifier
 	) -> void {
-		Modifier* mutable_modifier = modifier_manager.static_modifiers.get_item_by_identifier(name);
+		ret &= add_key_map_entry(
+			key_map, modifier.get_identifier(), ONE_EXACTLY,
+			expect_dictionary(modifier_manager.expect_base_country_modifier(modifier))
+		);
+	};
 
-		if (mutable_modifier != nullptr) {
-			if (multiplier != fixed_point_t::_1()) {
-				*mutable_modifier *= multiplier;
+	const auto set_country_event_modifier = [&key_map, &modifier_manager, &ret](
+		Modifier const*& modifier_ptr, std::string_view const&& identifier, IconModifier::icon_t const& icon
+	) -> void {
+		ret &= add_key_map_entry(
+			key_map, identifier, ONE_EXACTLY,
+			[&modifier_manager, &modifier_ptr, &identifier, &icon]
+			(ast::NodeCPtr const& value)->bool {
+				modifier_ptr = modifier_manager.get_event_modifier_by_identifier(identifier);
+				if(modifier_ptr != nullptr) {
+					return true; //an event modifier overrides a static modifier with the same identifier.
+				}
+
+				ModifierValue modifier_value {};
+				bool has_parsed_modifier = expect_dictionary(modifier_manager.expect_base_country_modifier(modifier_value))(value);
+				has_parsed_modifier &= modifier_manager.add_event_modifier(identifier, std::move(modifier_value), icon, STATIC);
+				if(has_parsed_modifier) {
+					modifier_ptr = modifier_manager.get_event_modifier_by_identifier(identifier);
```

Comment:
It would be faster to just get the back modifier from the event registry's internal data deque.

Reply:
It feels wrong for the StaticModifierCache to access the internals of the modifier_manager's event_modifiers storage imo.

Reply:
You don't have to use anything private, just `&modifier_manager.get_event_modifiers().back()`. It's the standard way we get a just-added element across the codebase.